.github/workflows/lint.yml

```diff
@@ -0,0 +1,28 @@
+name: Lint python files changed by a PR
+
+on:
+  pull_request:
+    branches: [master]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ["3.11"]
```

Comment:
No particular reason. I don't think it really matters. This is just the version of python that will host the linting tools. I went with 3.11 because it's stable and relatively future-proof, but I'm just as happy to change it if there's a reason to.

I don't see a need to run the linting tools against multiple versions of python (they're static analyzers).